Budget app

AmineDerbal opened the pull request:
In this pull request ♻️ 

- Create a budget app that registers and log in and keeps track of all expenses in the categories 👍  
- Rename the entity table to expenses 👍 
- Create an intermediary table between expenses and groups to solve the many to many relationship 👍 
-  Secure app from n+1 problems by using through relationship in the model 👍 
- Create categories and expenses pages along with the forms to add new category and expense 👍 
- when adding an expense you can select either one or several categories 👍 
- The icons for the group table are saved in the assets/images folder 👍 
- Create units and integrations tests for the pages 👍 
- presentation video : https://www.loom.com/share/75541b5f79bc4c3ea7701da3b5185450?sid=d8124c46-4952-4244-9030-536f7e76e65b

AmineDerbal committed:
Create migrations and generate tables

AmineDerbal committed:
Create table models

AmineDerbal committed:
Create tables controllers

AmineDerbal committed:
Add devise gem

AmineDerbal committed:
Install devise

AmineDerbal committed:
Config devise parameters

AmineDerbal committed:
Migrate users table to add authentification

AmineDerbal committed:
Migrate user table

AmineDerbal committed:
Update config for devise

AmineDerbal committed:
Add canacnacn gem

AmineDerbal committed:
config authorizations

AmineDerbal committed:
Add bootstrap

AmineDerbal committed:
Add Home controller

AmineDerbal committed:
Add home route as the root

AmineDerbal committed:
Generate devise views

AmineDerbal committed:
Add Proxima Nova font

AmineDerbal committed:
Add arrow icon

AmineDerbal committed:
Config parameters

AmineDerbal committed:
Update the login and registration page

AmineDerbal committed:
Add css styling

AmineDerbal committed:
Add route for groups as categories

AmineDerbal committed:
Create Home views and controller

AmineDerbal committed:
Fix typos in the model

AmineDerbal committed:
Setup the sign out method

AmineDerbal committed:
Create a header partial

AmineDerbal committed:
Update groups controller

AmineDerbal committed:
Create views for groups

AmineDerbal committed:
Update the styling for the app

AmineDerbal committed:
Add routing for expenses

AmineDerbal committed:
Fix expense model

AmineDerbal committed:
Update the header view

AmineDerbal committed:
Update the category index

AmineDerbal committed:
Update the category form

AmineDerbal committed:
Add view methods in the controller

AmineDerbal committed:
Create index view for expense page

AmineDerbal committed:
Create form to add expenses

AmineDerbal committed:
update the styling

AmineDerbal committed:
Add rspec gem

AmineDerbal committed:
install rspec

AmineDerbal committed:
Config rspec for testing

AmineDerbal committed:
Fix stylelint errors

AmineDerbal committed:
Update files to fix rubocop errors

AmineDerbal committed:
Update view

AmineDerbal committed:
Create unit tests for categories

AmineDerbal committed:
Create unit tests for expenses

AmineDerbal committed:
Add integration tests for categories

AmineDerbal committed:
Update test

AmineDerbal committed:
Create integration tests for expenses

AmineDerbal committed:
Add License

AmineDerbal committed:
update readme

AmineDerbal committed:
Config deployement

AmineDerbal committed:
Update config

AmineDerbal committed:
Update readme

AmineDerbal committed:
Add icon for logout

AmineDerbal committed:
update readme

elisha2kyakpo1 commented on README.md:
```diff
@@ -1,24 +1,195 @@
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [You can check the live demo here](https://budget-expense-app.onrender.com/)
```
- [ ] I got an error when I tried to register on the app, kindly fix the error in production for the app to work as expected.

![live](https://github.com/AmineDerbal/Budget-App-Rails-Capstone-Project/assets/53124548/9742382f-db18-4cce-b3ca-ab3caab01282)

elisha2kyakpo1 commented on README.md:
```diff
@@ -1,24 +1,195 @@
+
+## 📺 Presentation <a name="presentation"></a>
+- [Link to video presentation](https://www.loom.com/share/75541b5f79bc4c3ea7701da3b5185450?sid=18c2c3e0-870c-4904-8221-5c065014e947)
+
```
- [ ] Great job recording a video as per requirement, unfortunately, the video doesn't have sound. Kindly record another video and follow all the video requirements.

![video](https://github.com/AmineDerbal/Budget-App-Rails-Capstone-Project/assets/53124548/275f584b-89c0-43a0-a43c-a9c8856a30b9)

elisha2kyakpo1 commented on app/views/groups/index.html.erb:
```diff
@@ -0,0 +1,27 @@
+  <% end %>
+
+  <%= link_to "ADD CATEGORY", new_group_path, method: :get, class: "btn", id:'add-category' %>
+</div>
```
- [ ] Kindly make sure that the design matches the given design, this means that spaces between elements, colors, typography, fonts, etc. should match.

![cat](https://github.com/AmineDerbal/Budget-App-Rails-Capstone-Project/assets/53124548/172749a9-a469-47d2-ad23-5dae72fa634b)

![exp](https://github.com/AmineDerbal/Budget-App-Rails-Capstone-Project/assets/53124548/77e74652-49b3-41fa-b1a9-8fcfed1fa3dc)


![design](https://github.com/AmineDerbal/Budget-App-Rails-Capstone-Project/assets/53124548/e2ff9793-cb46-4b1a-8fd1-a212cc2a94d9)

elisha2kyakpo1 reviewed (changes requested):
Hi 👋 @AmineDerbal, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Tests passing👏
👏 The app works as expected on the local machine.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

AmineDerbal committed:
update production config

AmineDerbal committed:
Update production

AmineDerbal committed:
update views

AmineDerbal committed:
Update production config

AmineDerbal committed:
Update production config

AmineDerbal committed:
Update production config

AmineDerbal committed:
fix the spacing and typography

AmineDerbal committed:
update readme

AmineDerbal commented:
- Fixed the views to match the design 👍 
- Fixed the production error in the deployment 👍 
- On the deployed page once registered you will receive a confirmation email (check your spam folder in case) 👍 
- the new presentation video https://www.loom.com/share/af2e66d770d445999b3478945981f9a4?sid=03dd5e79-9193-46f4-8102-240b26c7d969 👍 
- for the deployment the email address should be valid 👍

AmaduKamara reviewed (approved):
# Status: Approved 🟢 🟢 🟢

Hi @AmineDerbal,

Good job working on the changes highlighted by the previous reviewer 👍🏽

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ The app works fine both locally and online 👍🏽
✅ Good video presentation 👍🏽
✅ Good commit messages 👍🏽
✅ Your README is professional 👍🏽

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```
